src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py

```diff
@@ -3,24 +3,477 @@
+
+        for key in _coalesce_lst(alt_rsa_token_keys):
+            alternate_keys.append(_rsa_token_key_factory(key))
+
+        for key in _coalesce_lst(alt_x509_certificate_token_keys):
+            alternate_keys.append(_x509_token_key_factory(key))
+
+        if token_claims is not None:
+            for key in token_claims:
+                claim = ContentKeyPolicyTokenClaim(claim_type=key,
+                                                   claim_value=token_claims[key])
+                _token_claims.append(claim)
+
+        restriction = ContentKeyPolicyTokenRestriction(
+            issuer=issuer, audience=audience, primary_verification_key=primary_verification_key,
+            alternate_verification_keys=alternate_keys, required_claims=_token_claims,
+            restriction_token_type=token_type,
+            open_id_connect_discovery_document=open_id_connect_discovery_document)
+
+    if restriction is None or configuration is None:
+        raise CLIError(
+            'Could not build content key policy option due to invalid restriction or configuration.')
+
+    return ContentKeyPolicyOption(name=policy_option_name,
+                                  configuration=configuration,
+                                  restriction=restriction)
+
+
+# Utility functions
+# Returns string if not null, or an empty string otherwise.
+def _coalesce_lst(value):
+    return value or []
+
+
+def _coalesce_timedelta(value):
+    return parse_timedelta(value) if value else None
+
+
+def _read_json(path):
+    return _read(path, 'r')
+
+
+def _read_binary(path):
+    return _read(path, 'rb')
+
+
+def _read(path, read_type):
+    with open(path, read_type) as file:
+        return file.read()
+
+
+def _b64_to_str(data):
+    return base64.b64encode(data)
+
+
+def _int2bytes(int_value):
+    hex_value = '{0:x}'.format(int_value)
+    # make length of hex_value a multiple of two
+    hex_value = '0' * (len(hex_value) % 2) + hex_value
+    return codecs.decode(hex_value, 'hex_codec')
+
+
+# Counts how many values are truthy on a list.
+def _count_truthy(values):
+    return len([value for value in values if value])
+
+
+# Factories
+def _symmetric_token_key_factory(input_symmetric):
+    key = bytearray(input_symmetric, 'utf-8')
+    return ContentKeyPolicySymmetricTokenKey(key_value=key)
+
+
+def _rsa_token_key_factory(input_rsa):
+    content = _read(input_rsa, 'r')
+    rsa_key = serialization.load_pem_public_key(
+        content.encode('ascii'),
+        backend=default_backend()
+    )
+
+    exp = rsa_key.public_numbers().e
+    mod = rsa_key.public_numbers().n
+
+    return ContentKeyPolicyRsaTokenKey(
+        exponent=bytearray(_int2bytes(exp)), modulus=bytearray(_int2bytes(mod)))
+
+
+def _x509_token_key_factory(input_x509):
+    content = _read(input_x509, 'r')
+    return ContentKeyPolicyX509CertificateTokenKey(
+        raw_body=bytearray(content, 'ascii'))
+
+
+def _play_ready_configuration_factory(content):
```

Comment:
Try parsin the JSON with PlayReady(**play_ready_template)

Reply:
It did not work because we also need to handle duration objects, and kwargs won't really be able to realize it's needed hence